e2e: test Overlap consistent failure

Test failure:
```
Failed
=== RUN   TestOverlap
    overlap_test.go:71: 1500/2000 retries reached for github.com/hashicorp/nomad/e2e/overlap.TestOverlap (err=timed out before an allocation was found for overlap6f928495)
    overlap_test.go:71: timeout: timed out before an allocation was found for overlap6f928495
--- FAIL: TestOverlap (24.85s)
```

--- a/e2e/overlap/overlap_test.go
+++ b/e2e/overlap/overlap_test.go
@@ -20,54 +20,24 @@ func TestOverlap(t *testing.T) {
 	nomadClient := e2eutil.NomadClient(t)
 	e2eutil.WaitForLeader(t, nomadClient)
 
-	// Wait for at least 1 feasible node to be ready and get its ID
-	var node *api.Node
-	testutil.Wait(t, func() (bool, error) {
-		nodesList, _, err := nomadClient.Nodes().List(nil)
-		if err != nil {
-			return false, fmt.Errorf("error listing nodes: %v", err)
-		}
-
-		for _, n := range nodesList {
-			if n.Status != "ready" {
-				continue
-			}
-			if n.SchedulingEligibility != "eligible" {
-				continue
-			}
-
-			node, _, err = nomadClient.Nodes().Info(n.ID, nil)
-			must.NoError(t, err)
-
-			if node.Attributes["kernel.name"] != "linux" {
-				continue
-			}
-
-			return true, nil
-		}
-
-		return false, fmt.Errorf("no nodes ready before timeout; need at least 1 ready")
-	})
-
-	// Force job to fill one exact node
 	getJob := func() (*api.Job, string) {
 		job, err := e2eutil.Parse2(t, "testdata/overlap.nomad")
 		must.NoError(t, err)
 		jobID := *job.ID + uuid.Short()
 		job.ID = &jobID
-		job.Datacenters = []string{node.Datacenter}
-		job.Constraints[1].RTarget = node.ID
-		availCPU := int(node.NodeResources.Cpu.CpuShares - int64(node.ReservedResources.Cpu.CpuShares))
-		job.TaskGroups[0].Tasks[0].Resources.CPU = &availCPU
 		return job, *job.ID
 	}
 	job1, jobID1 := getJob()
 
+	// Register initial job that should block subsequent job's placement until
+	// its shutdown_delay is up.
 	_, _, err := nomadClient.Jobs().Register(job1, nil)
 	must.NoError(t, err)
 	defer e2eutil.WaitForJobStopped(t, nomadClient, jobID1)
 
+	// Capture node so we can ensure 2nd job is blocked by first
 	var origAlloc *api.AllocationListStub
+	var node *api.Node
 	testutil.Wait(t, func() (bool, error) {
 		a, _, err := nomadClient.Jobs().Allocations(jobID1, false, nil)
 		must.NoError(t, err)
@@ -77,8 +47,14 @@ func TestOverlap(t *testing.T) {
 		must.Len(t, 1, a)
 
 		origAlloc = a[0]
-		return origAlloc.ClientStatus == "running", fmt.Errorf("timed out before alloc %s for %s was running: %s",
-			origAlloc.ID, jobID1, origAlloc.ClientStatus)
+		if origAlloc.ClientStatus != "running" {
+			return false, fmt.Errorf("timed out before alloc %s for %s was running: %s",
+				origAlloc.ID, jobID1, origAlloc.ClientStatus)
+		}
+
+		node, _, err = nomadClient.Nodes().Info(origAlloc.NodeID, nil)
+		must.NoError(t, err)
+		return true, nil
 	})
 
 	// Stop job but don't wait for ClientStatus terminal
@@ -94,9 +70,13 @@ func TestOverlap(t *testing.T) {
 			a.ID, ds, cs)
 	})
 
-	// Start replacement job and assert it is blocked
+	// Start replacement job on same node and assert it is blocked
 	job2, jobID2 := getJob()
+	job2.Constraints = append(job2.Constraints, api.NewConstraint("${node.unique.id}", "=", origAlloc.NodeID))
 	job2.TaskGroups[0].Tasks[0].ShutdownDelay = 0 // no need on the followup
+	availCPU := int(node.NodeResources.Cpu.CpuShares - int64(node.ReservedResources.Cpu.CpuShares))
+	job2.TaskGroups[0].Tasks[0].Resources.CPU = &availCPU // require job1 to free resources
+
 	resp, _, err := nomadClient.Jobs().Register(job2, nil)
 	must.NoError(t, err)
 	defer e2eutil.WaitForJobStopped(t, nomadClient, jobID2)
@@ -113,8 +93,8 @@ func TestOverlap(t *testing.T) {
 	// Wait for job1's ShutdownDelay for origAlloc.ClientStatus to go terminal
 	sleepyTime := minStopTime.Sub(time.Now())
 	if sleepyTime > 0 {
-		t.Logf("Sleeping for the rest of the shutdown_delay (%.3s/%s)",
-			sleepyTime, job1.TaskGroups[0].Tasks[0].ShutdownDelay)
+		t.Logf("Followup job %s blocked. Sleeping for the rest of %s's shutdown_delay (%.3s/%s)",
+			*job2.ID, *job1.ID, sleepyTime, job1.TaskGroups[0].Tasks[0].ShutdownDelay)
 		time.Sleep(sleepyTime)
 	}
 

--- a/e2e/overlap/overlap_test.go
+++ b/e2e/overlap/overlap_test.go
@@ -35,9 +35,7 @@ func TestOverlap(t *testing.T) {
 	must.NoError(t, err)
 	defer e2eutil.WaitForJobStopped(t, nomadClient, jobID1)
 
-	// Capture node so we can ensure 2nd job is blocked by first
 	var origAlloc *api.AllocationListStub
-	var node *api.Node
 	testutil.Wait(t, func() (bool, error) {
 		a, _, err := nomadClient.Jobs().Allocations(jobID1, false, nil)
 		must.NoError(t, err)
@@ -47,15 +45,13 @@ func TestOverlap(t *testing.T) {
 		must.Len(t, 1, a)
 
 		origAlloc = a[0]
-		if origAlloc.ClientStatus != "running" {
-			return false, fmt.Errorf("timed out before alloc %s for %s was running: %s",
-				origAlloc.ID, jobID1, origAlloc.ClientStatus)
-		}
-
-		node, _, err = nomadClient.Nodes().Info(origAlloc.NodeID, nil)
-		must.NoError(t, err)
-		return true, nil
+		return origAlloc.ClientStatus == "running", fmt.Errorf("timed out before alloc %s for %s was running: %s",
+			origAlloc.ID, jobID1, origAlloc.ClientStatus)
 	})
+
+	// Capture node so we can ensure 2nd job is blocked by first
+	node, _, err := nomadClient.Nodes().Info(origAlloc.NodeID, nil)
+	must.NoError(t, err)
 
 	// Stop job but don't wait for ClientStatus terminal
 	_, _, err = nomadClient.Jobs().Deregister(jobID1, false, nil)

--- a/e2e/overlap/testdata/overlap.nomad
+++ b/e2e/overlap/testdata/overlap.nomad
@@ -1,15 +1,10 @@
 job "overlap" {
-  datacenters = ["dc1"]
+  datacenters = ["dc1", "dc2"]
   type        = "service"
 
   constraint {
     attribute = "${attr.kernel.name}"
     value     = "linux"
-  }
-
-  constraint {
-    attribute = "${node.unique.id}"
-    value     = "<<Must be filled in by test>>"
   }
 
   group "overlap" {
@@ -27,8 +22,7 @@ job "overlap" {
       }
 
       resources {
-        # Must be filled in by test
-        cpu    = "0"
+        cpu    = "500"
         memory = "50"
       }
     }

